src/Data/Text/Builder/Linear/Char.hs

```diff
@@ -72,3 +90,99 @@ unsafePrependCharM marr off c = case utf8Length c of
+-- Multiple chars
+--------------------------------------------------------------------------------
+
+-- | Prepend a given count of a 'Char' to a 'Buffer'.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> prependChars 3 'x' (b |>. 'A'))
+-- "xxxA"
+prependChars ∷ Word → Char → Buffer ⊸ Buffer
+prependChars count ch buff
+  | count == 0 = buff
+  | isAscii ch = prependExact
+      (fromIntegral count)
+      (\dst dstOff → unsafeReplicate dst dstOff (fromIntegral count) (ord ch))
+      buff
+  | otherwise =
+      (\cLen →
+        (\totalLen → prependExact
+          totalLen
+          (\dst dstOff → unsafeWrite dst dstOff ch *> unsafeTile dst dstOff totalLen cLen)
+          buff)
+        (cLen * fromIntegral count))
+      (utf8Length ch)
+
+-- | Apppend a given count of a 'Char' to a 'Buffer'.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> appendChars 3 'x' (b |>. 'A'))
+-- "Axxx"
+appendChars ∷ Word → Char → Buffer ⊸ Buffer
+appendChars count ch buff
+  | count == 0 = buff
+  | isAscii ch = appendExact
+      (fromIntegral count)
+      (\dst dstOff → unsafeReplicate dst dstOff (fromIntegral count) (ord ch))
+      buff
+  | otherwise =
+      (\cLen →
+        (\totalLen → appendExact
+          totalLen
+          (\dst dstOff → unsafeWrite dst dstOff ch *> unsafeTile dst dstOff totalLen cLen)
+          buff)
+        (cLen * fromIntegral count))
+      (utf8Length ch)
+
+--------------------------------------------------------------------------------
+-- Padding
+--------------------------------------------------------------------------------
+
+-- | Pad a builder from the /left/ side to the specified length with the specified
+-- character.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> justifyRight 10 'x' (appendChars 3 'A' b))
+-- "xxxxxxxAAA"
+-- >>> runBuffer (\b -> justifyRight 5 'x' (appendChars 6 'A' b))
+-- "AAAAAA"
+justifyRight ∷ Word → Char → Buffer ⊸ Buffer
+justifyRight n ch buff = case lengthOfBuffer buff of
+  (# buff', len #) → case Linear.move len of
+    Linear.Ur len' → if n <= len'
+      then buff'
+      else prependChars (n - len') ch buff'
+
+-- | Pad a builder from the /right/ side to the specified length with the specified
+-- character.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> justifyLeft 10 'x' (appendChars 3 'A' b))
+-- "AAAxxxxxxx"
+-- >>> runBuffer (\b -> justifyLeft 5 'x' (appendChars 6 'A' b))
+-- "AAAAAA"
+justifyLeft ∷ Word → Char → Buffer ⊸ Buffer
+justifyLeft n ch buff = case lengthOfBuffer buff of
+  (# buff', len #) → case Linear.move len of
+    Linear.Ur len' → if n <= len'
+      then buff'
+      else appendChars (n - len') ch buff'
+
+-- | Center a builder to the specified length with the specified character.
+--
+-- >>> :set -XLinearTypes
+-- >>> runBuffer (\b -> center 10 'x' (appendChars 3 'A' b))
+-- "xxxxAAAxxx"
+-- >>> runBuffer (\b -> center 5 'x' (appendChars 6 'A' b))
+-- "AAAAAA"
+center ∷ Word → Char → Buffer ⊸ Buffer
```

Comment:
You mean the padding functions specifically? Mainly pretty-printing. For example, I generate some C files with lots of arrays, but I want them to be well presented to facilitate review, so padding with 0 for hex values and spaces for general text comes quite handy.

This also bring the API closer to the `text` one.

Reply:
Could you possibly share a snippet using the proposed API? What happens when you already have some data in a `Buffer` and want to append a padded hexadecimal?

Reply:
After try using the API with non-trivial examples, I see what you mean 😄

The immediate solution I came with is to add the following function to the API:

```haskell
newEmptyBuffer ∷ Buffer ⊸ (# Buffer, Buffer #)
newEmptyBuffer (Buffer t@(Text arr _ _)) =
  (# Buffer t, Buffer (if isPinned arr then memptyPinned else mempty) #)
```

then an example of the use of `justifyRight` is:

```haskell
-- | Convenient function to append padded hex
infixl 6 |>&&
(|>&&) :: (Integral a, FiniteBits a) => Buffer ⊸ (Word, a) -> Buffer
b |>&& (w, n) = case newEmptyBuffer b of
  (# b', empty #) -> (b' |># "0x"#) >< justifyRight w '0' (empty |>& n)

-- Example
runBuffer \b -> (b |># "Foo "#) |>&& (4, 0xf :: Word) |># "; "# |>&& (4, 0xabc :: Word)
-- "Foo 0x000f; 0x0abc"
```

Reply:
Yeah, let's add `newEmptyBuffer` than. Not sure about the name though...

Reply:
Done. About naming: I did not named simply `emptyBuffer` because it requires an input. But I have not strong opinion about this. Other candidates: `makeEmptyBuffer` or the shorter `mkEmptyBuffer`.